### PR TITLE
repetier-server

### PR DESCRIPTION
**Pull request by Lukas1818**
#359 Add support for the printer server [reptier-server](https://www.repetier-server.com/).

This is a merge of https://github.com/prusa3d/PrusaSlicer/pull/4384

### CODE REVIEW

**Comment by supermerill** [comment]
I'm wondering how this will play with the printer-config rework prusa has done on 2.3

File: src/libslic3r/PrintConfig.cpp [hunk 1]
```diff
@@ -148,7 +148,14 @@ void PrintConfigDef::init_common_params()
                    "the API Key or the password required for authentication.");
     def->mode = comAdvanced;
     def->set_default_value(new ConfigOptionString(""));
-
+    
+    def = this->add("printhost_slug", coString);
+    def->label = L("Printer");
+    def->tooltip = L("Name of the printer");
+    def->gui_type = "select_open";
```
Comment (supermerill) [inline src/libslic3r/PrintConfig.cpp:155]: this line def->gui_type = "select_open"; makes the slicer crash.  It's unsupported and maybe un-needed (print_host use nothing).

**Comment by Lukas1818** [comment]
Sorry for the bug. I had tested the merge, but no crashes happened.
When does it happen?

**Comment by supermerill** [comment]
At GUI settings initialization

**Comment by Lukas1818** [comment]
I could not reproduce this behavior.

**Comment by supermerill** [comment]
yeah!
after fixing a bug in the old code
after just finishing merging it with the new physical printer for 2.3
now prusa has just announced their own integration, and that should replace all the previous work.
And the removal/replacement will take me a bit of time...
joy

**Pull request merged**